photutils/detection/core.py

```diff
@@ -295,39 +302,79 @@ def find_peaks(data, threshold, min_separation=2, exclude_border=True,
         detected peaks exceeds ``npeaks``, the peaks with the highest
         peak intensities will be returned.
 
-    footprint : `~numpy.ndarray` of bools, optional
-        A boolean array where `True` values describe the local footprint
-        region within which to search for peaks at every point in
-        ``data``.  Overrides ``min_separation``, except for border
-        exclusion when ``exclude_border=True``.
+    subpixel : bool, optional
+        If `True`, then a cutout of the specified ``box_size`` or
+        ``footprint`` will be taken centered on each peak and fit with a
+        2D Gaussian.  In this case, the fitted local centroid and peak
+        amplitude will also be returned in the output table.
 
     Returns
     -------
     output : `~astropy.table.Table`
         A table containing the x and y pixel location of the peaks and
-        their values.
+        their values.  If ``subpixel=True``, then the table will also
+        contain the local centroid and fitted peak value.
     """
 
-    if segment_image is not None:
-        if segment_image.shape != data.shape:
-            raise ValueError('segment_image and data must have the same '
-                             'shape')
-
-    from skimage.feature import peak_local_max
-    coords = peak_local_max(data, min_distance=int(min_separation),
-                            threshold_abs=threshold, threshold_rel=0.0,
-                            exclude_border=exclude_border, indices=True,
-                            num_peaks=npeaks, footprint=footprint,
-                            labels=segment_image)
-    y_peaks, x_peaks = coords[:, 0], coords[:, 1]
-    peak_values = data[y_peaks, x_peaks]
-
-    if coords.shape[0] > npeaks:
-        # NOTE: num_peaks is ignored by peak_local_max() if labels are input
+    from scipy import ndimage
+
+    if np.all(data == data.flat[0]):
+        return []
+
+    if footprint is not None:
+        data_max = ndimage.maximum_filter(data, footprint=footprint,
+                                          mode='constant', cval=0.0)
+    else:
+        data_max = ndimage.maximum_filter(data, size=box_size,
+                                          mode='constant', cval=0.0)
+
+    peak_data = data.copy()
+    peak_data *= (data == data_max)
+
+    if border_width is not None:
+        for i in range(peak_data.ndim):
+            peak_data = peak_data.swapaxes(0, i)
+            peak_data[:border_width] = 0.
+            peak_data[-border_width:] = 0.
```

Comment:
Here you use `0` as a special value ... what if my peak pixel has value 0 because all the rest of the image is negative?

As I said, I didn't test, but it looks to me this might not work in that case?